zrml/prediction-markets/src/lib.rs

```diff
@@ -1259,28 +1290,6 @@ mod pallet {
             };
 
             let market_type = market.market_type.clone();
-            if let MarketType::Categorical(_) = market.market_type {
-                if let Outcome::Categorical(index) = resolved_outcome {
-                    let assets = Self::outcome_assets(*market_id, market);
-                    total_categories = assets.len().saturated_into();
-                    for asset in assets.iter() {
-                        if let Asset::CategoricalOutcome(_, inner_index) = asset {
-                            if index == *inner_index {
-                                continue;
-                            }
-
-                            let (total_accs, accounts) = T::Shares::accounts_by_currency_id(*asset);
-                            if total_accounts == 0 {
-                                total_accounts = total_accs;
-                            }
-
-                            total_asset_accounts = total_asset_accounts
-                                .saturating_add(accounts.len().saturated_into());
-                            T::Shares::destroy_all(*asset, accounts.iter().cloned());
-                        }
-                    }
-                }
-            }
```

Comment:
Removal of loosing categorical assets now happens when calling the `Swap::set_pool_as_stale` function